src/mkdocs_autorefs/references.py

```diff
@@ -127,6 +173,9 @@ def _make_tag(self, identifier: str, text: str) -> Element:
             A new element.
         """
         el = Element("span")
+        if self.hook:
+            identifier = self.hook.expand_identifier(identifier)
+            el.attrib.update(self.hook.get_context().as_dict())
```

Comment:
issue: No way to preserve attribute order in the generated span, and autorefs' regex expects the identifier first


When building the span element, even if we set its attributes in a specific order, it seems they'll always get sorted in the generated HTML. For example:

```html
<span
    data-autorefs-domain="py"
    data-autorefs-filepath="some/filepath.py"
    data-autorefs-identifier="some.identifier"
    data-autorefs-lineno="42"
    data-autorefs-origin="id.of.docstring.parent.object"
    data-autorefs-role="class"
>...</span>
```

This is problematic because autorefs uses a regular expression to find cross-ref spans, and expects the `data-autorefs-identifier` attribute to always appear first.

I suggest again (see https://github.com/mkdocstrings/autorefs/issues/41#issuecomment-1961673253) that we use a custom tag instead:

```html
<autoref
    domain="py"
    filepath="some/filepath.py"
    identifier="some.identifier"
    lineno="42"
    origin="id.of.docstring.parent.object"
    role="class"
>...</autoref>
```

...so that we can easily find them with a regex, and parse the attributes with a reduced HTML parser:

```python
AUTO_REF_RE = re.compile(r"<autoref (?P<attrs>.*?)>(?P<title>.*?)</autoref>")

from html.parser import HTMLParser
class AttrsParser(HTMLParser):
    def __init__(self):
        super().__init__(self)
        self.attrs = []

    def parse(self, html):
        self.attrs.clear()
        self.feed(html)
        return self.attrs

    def handle_starttag(self, tag, attrs):
        self.attrs.extend(attrs)


# for each match, build f"<a {match.group("attrs")}></a>" and pass it to the parser
AttrsParser().parse('<a data-preview data-identifier="pathlib.Path" data-other="0">some title</a>')
# [('data-preview', None), ('data-identifier', 'pathlib.Path'), ('data-other', '0')]
```

Reply:
See #48.

Reply:
Benchmarks showed that the current approach is more efficient than a regex without groups and the HTML parser feeding on the whole match.